src/core/plugin/luapi_application.h

```diff
@@ -231,6 +293,48 @@ static int applib_getFilePath(lua_State* L) {
     return args_returned;
 }
 
+/**
+ * Create an 'Open File' dialog and when the user has chosen a filepath
+ * call a callback function whose sole argument is the filepath.
+ *
+ * @param cb string name of the callback function(path:string) to call when a file was chosen
+ * @param types string[] array of the different allowed extensions e.g. {'\*.bmp', '\*.png'}
+ *
+ * Examples:
+ *   app.fileDialogOpen("cb", {})
+ *   app.fileDialogOpen("cb", {'*.bmp', '*.png'})
+ */
+static int applib_fileDialogOpen(lua_State* L) {
+
+    // discard any extra arguments passed in
+    lua_settop(L, 2);
+    const char* callback = luaL_checkstring(L, 1);
```

Comment:
Great question!

According to the documentation `luaL_checkstring` uses [lua_tolstring](https://pgl.yoyo.org/luai/i/lua_tolstring) to get its result, so all conversions and caveats of that function apply here, and

> [lua_tolstring](https://pgl.yoyo.org/luai/i/lua_tolstring) returns a fully aligned pointer to a string inside the Lua state. This string always has a zero ('\0') after its last character (as in C), but can contain other zeros in its body. Because Lua has garbage collection, there is no guarantee that the pointer returned by [lua_tolstring](https://pgl.yoyo.org/luai/i/lua_tolstring) will be valid after the corresponding value is removed from the stack.

Removal from the stack should happen after the function `applib_fileDialogOpen` returns, which is after the dialog has been constructed but before the callback is executed.
The stack is created at the time the `fileDialogOpen` function is called from Lua.

As I understand the lambda passed to `xoj::OpenDlg` captures the string by value, but that is only the value of the pointer, not the string, right?!

So it looks to me that the string could be garbage collected from the Lua side after the dialog shows up at any time. I didn't manage to make it malfunction in some way, even when forcing garbage collection on the Lua side. The callback was always correctly executed.